src/interpreter.rs

```diff
@@ -110,17 +110,15 @@ impl<'a> Evaluator<'a> {
                 input_mapping.insert(*symbol_id, *input_ref);
             }
 
-            // Modified code for outputs to handle None cases
-            if let Some(output_ref) = sys
+            // Map outputs by matching symbol name to sys.outputs' names
+            if let Some((idx, _)) = sys
                 .outputs
                 .iter()
-                .filter_map(|o| {
-                    // Only process outputs that have a name
-                    ctx.get_symbol_name((o).expr).map(|name| (o, name))
-                })
-                .find(|(_, name)| *name == symbol_name)
+                .map(|o| ctx[o.name].to_string())
+                .enumerate()
+                .find(|(_, out_name)| out_name == symbol_name)
             {
-                output_mapping.insert(*symbol_id, *output_ref.0);
+                output_mapping.insert(*symbol_id, sys.outputs[idx]);
```

Comment:
@Nikil-Shyamsunder There might be a better way to do this.

The old code was decent, with the one problem that `get_symbol_name` does not make sense for outputs, since outputs are rarely symbols but more complex expressions instead. Instead, you want to look at the `o.name` as you are doing in the new structure.

Here is a sketch for solving this more elegantly:
```.rs
let output = sys.outputs.iter().find(|o| ctx[o.name] == symbol_name).expect("failed to find matching output");
output_mapping.insert(*symbol_id, output);
```